source/distributed/tria.cc

```diff
@@ -1502,16 +1501,11 @@ namespace parallel
             first_vertices.resize(cells);
             for (unsigned int c=0; c<cells; ++c)
               {
-                // We need to store a pointer to the first vertex.
-                const dealii::Point<spacedim> *const vertex
-                  = reinterpret_cast<const dealii::Point<spacedim> *>(ptr);
                 first_indices[c] = vertices.size();
-                vertices.push_back(*vertex);
-                ptr += sizeof(dealii::Point<spacedim>);
-                vertices.resize(vertices.size() + n_vertices_on_cell[c]-1);
-                memcpy(&vertices[vertices.size() - (n_vertices_on_cell[c]-1)],
-                       ptr, (n_vertices_on_cell[c]-1)*sizeof(dealii::Point<spacedim>));
-                ptr += (n_vertices_on_cell[c]-1)*sizeof(dealii::Point<spacedim>);
+                vertices.resize(vertices.size() + n_vertices_on_cell[c]);
+                memcpy(&vertices[vertices.size() - n_vertices_on_cell[c]],
+                       ptr, n_vertices_on_cell[c]*sizeof(dealii::Point<spacedim>));
+                ptr += n_vertices_on_cell[c]*sizeof(dealii::Point<spacedim>);
```

Comment:
It's unclear to me what exactly the author of that code was trying to achieve by first copying one element, and then all of the others. Good catch!